# Data Analysis:
This notebook aims to make an initial description of the data available for training.

In [2]:
import pandas as pd

In [3]:
meta = pd.read_csv('data/TrainingData_BU&Public_CWS.csv')

/var/folders/hy/36n7jr6x3h74ddx507dc12_00000gn/T/ipykernel_36898/1568081312.py:1: DtypeWarning: Columns (22) have mixed types. Specify dtype option on import or set low_memory=False.
  meta = pd.read_csv('data/TrainingData_BU&Public_CWS.csv')


In [4]:
meta.recording_id.value_counts().head()

104940.0    394
104939.0    370
104942.0    314
104941.0    290
72961.0     286
Name: recording_id, dtype: int64

The recording ids aren't unique, indicating that there are mulitple references to the same recording within the database.  

In [5]:
len(meta.project.unique())

114

There are 114 projects in the database. 

In [6]:
len(meta.recording_id.unique())

97439

There are 97439 unique recordings in the database. 

In [7]:
len(meta[meta.clip_url.isna()])

28386

Some of these clips don't have clip_url links to an audio file. There are a total of 28386 un-linked tags out of a total 1152840 tags. 